xarray/core/variable.py

```diff
@@ -492,6 +495,46 @@ def _broadcast_indexes_advanced(self, key):
         key = VectorizedIndexer(variable.data for variable in variables)
         return dims, key
 
+    def _broadcast_indexes_dask_pointwise(self, key):
+        if any(not isinstance(k, (Variable, slice)) for k in key):
```

Comment:
Hehe good :). Ah yeah and vindex is simpler (unless you support boolean arrays being orthogonal).